BASE/src/Microsoft.ApplicationInsights/Extensibility/Implementation/Tracing/SelfDiagnosticsProvider.cs

```diff
@@ -0,0 +1,71 @@
+﻿namespace Microsoft.ApplicationInsights.Extensibility.Implementation.Tracing
+{
+    using System;
+    using System.Collections.Generic;
+
+    using Microsoft.ApplicationInsights.Extensibility.Implementation.ConfigString;
+
+    /// <summary>
+    /// This class encapsulates parsing and interpreting the self diagnostics configuration string.
+    /// </summary>
+    internal class SelfDiagnosticsProvider
+    {
+        private const string KeyDestination = "Destination";
+        private const string KeyFilePath = "Path";
+        private const string KeyLevel = "Level";
+        private const string KeyFileMaxSize = "MaxSize";
+        private const string ValueDestinationFile = "file";
+
+        /// <summary>
+        /// Parse a configuration string and return a Dictionary.
+        /// </summary>
+        /// <remarks>Example: "key1=value1;key2=value2;key3=value3".</remarks>
+        /// <returns>A dictionary parsed from the input configuration string.</returns>
+        internal static Dictionary<string, string> ParseConfigurationString(string configurationString)
+        {
+            var keyVaulePairs = ConfigStringParser.Parse(configurationString, configName: "Self-Diagnostics Configuration String");
+
+            if (keyVaulePairs.ContainsKey(KeyDestination))
+            {
+                return keyVaulePairs;
+            }
+            else
+            {
+                throw new Exception("Self-Diagnostics Configuration string is invalid. Missing key 'Destination'");
+            }
+        }
+
+        /// <summary>
+        /// Evaluates if the keyvalue pairs specifies writing to file. If so, parses and returns params.
+        /// </summary>
+        /// <param name="keyValuePairs">The keyvalue pairs from the config string.</param>
+        /// <param name="path">File directory for logging.</param>
+        /// <param name="level">Log level.</param>
+        /// <param name="maxSize">Max size of file log.</param>
```

Comment:
its best to be explicit about what units the size is described - KB? or just bytes? ServerTelemetryChannel has some size similar to this - you can borrow from there, if needed.

Reply:
i removed the maxSize since we can't support it at this time